src/utils/watcher.rs

```diff
@@ -130,8 +149,12 @@ mod tests {
 
         let (tx, rx) = sync_channel(1);
         let (started_tx, started_rx) = mpsc::sync_channel(1);
-        let _watcher =
-            Watcher::with_start_notification(config_path.clone(), |_| (), tx, Some(started_tx));
+        let _watcher = Watcher::with_start_notification(
```

Comment:
Could you add a test for `ConfigPath::Regular` for the case of watching system_path, then creating user_path and watching that? (And then perhaps removing user_path and back to watching system_path)